src/sentry/api/event_search.py

```diff
@@ -883,14 +883,16 @@ def get_reference_event_conditions(organization, snuba_args, event_slug):
 
     This is a key part of pagination in the event details modal and
     summary graph navigation.
+
+    :deprecated: To be replaced with sentry.snuba.discover.create_reference_event_conditions
     """
     groupby = snuba_args.get("groupby", [])
     columns = get_columns_from_aliases(groupby)
     field_names = [get_snuba_column_name(field) for field in groupby]
 
     # Fetch the reference event ensuring the fields in the groupby
     # clause are present.
-    event_data = find_reference_event(organization, snuba_args, event_slug, columns)
+    event_data = find_reference_event(organization, event_slug, columns)
```

Comment:
Not directly related to this PR but the way we fetch the event data is not really ideal since `snuba_data` shouldn't be a public method. (https://github.com/getsentry/sentry/blob/master/src/sentry/api/event_search.py#L880)
Could we move away from getting the event_data like this?

Reply:
I think raw query makes sense in this case (eventstore is for when we want the actual event instance)

Reply:
Thanks for fixing